tools/python/boutiques/schema/examples/example1/exampleTool1.py

```diff
@@ -40,12 +40,17 @@ def main(args=None):
     if (not results.number) and (not results.enum_input):
         sys.stderr.write("error: one of number or enum_input is required.\n")
         sys.exit(1)
-    
+   
     var = os.getenv("ENVAR")
     if var != "theValue":
         sys.stderr.write("error: ENVAR environment variable must be set to 'theValue'.\n")
         sys.exit(1)
 
+    hostname = os.getenv("HOSTNAME")
+    if hostname != "testHost":
```

Comment:
My issue right now is that this error gets thrown during the Singularity test since ```example1_sing.json``` doesn't set the hostname (which doesn't seem to be possible with Singularity). Is there a way to check in ```exampleTool1.py1``` whether it's being used by Docker or Singularity?

Reply:
I think a Singularity image must contain `/.singularity.d/runscript` but @gkiar would know this better.

The additional 5 commits are typos that we agreed to push directly to master... but we probably shouldn't have and we won't do it again! 

Thanks for the PR btw, it looks good!

Reply:
Hey @glatard! @erinb90 and I just poked at this together, and will settle for setting & checking the `-e HOME=/tmp/somemadeuppath/`/`--home /tmp/somemadeuppath/` param in Docker/Singularity as a way to test this feature. We'll have to create the directory first, though, in the test. Thoughts?

Reply:
Sounds great, the directory could be created in the Dockerfile used by example1.py

Reply:
I created a directory `/tmp/testDir/` in the Dockerfile, went in the container and saw it exists. However, when the Singularity test is run with the `--home /tmp/testDir/` option, I get the error `Home directory is not owned by calling user`. So I tried `--home $HOME:/tmp/testDir/` to map the source home to the destination home (see my latest changes), but am now getting a weird error:
```
WARNING: Could not chdir to home: /tmp/testDir/
/.exec: line 3: /home/travis/build/boutiques/boutiques/tools/python/temp-566401995773-1537479246275.localExec.boshjob.sh: No such file or directory
```
The temp file is in the run command, why would it not exist?